packages/shell-api/src/enums.ts

```diff
@@ -26,8 +26,11 @@ export enum DBQueryOption {
 }
 
 export const shellApiType = Symbol.for('@@mongosh.shellApiType');
-// TODO: Would require changes to java-shell
-// export const asPrintable = Symbol('asPrintable');
-export const asShellResult = Symbol.for('@@mongosh.asShellResult');
+// TODO: This requires changes to java-shell.
+export const asPrintable = Symbol.for('@@mongosh.asPrintable');
+export const namespaceInfo = Symbol.for('@@mongosh.namespaceInfo');
+
+/// @deprecated toShellResult() returns both the raw value and the printable version now.
```

Comment:
same here, would not bother deprecating and just try to make the change in the java shell, then drop a message in mongosh-jetbrains.